From 607f9887ca85f246e970778e3d40aa5c346365a7 Mon Sep 17 00:00:00 2001
From: Artem Savkov <asavkov@redhat.com>
Date: Tue, 28 Sep 2021 18:38:48 +0200
Subject: [PATCH] Fix empty gateway->path with glib > 2.68

Starting with glib 2.69 g_match_info_fetch() returns an empty string
instead of NULL in case when no match was found. Properly handle this
case.

Signed-off-by: Artem Savkov <asavkov@redhat.com>
---
 src/discord-http.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/discord-http.c b/src/discord-http.c
index 83b92ce..80b3946 100644
--- a/src/discord-http.c
+++ b/src/discord-http.c
@@ -175,6 +175,9 @@ static void discord_http_gateway_cb(struct http_request *req)
 
     if (dd->gateway->path == NULL) {
       dd->gateway->path = g_strdup("/?encoding=json&v=6");
+    } else if (g_strcmp0(dd->gateway->path, "") == 0) {
+      g_free(dd->gateway->path);
+      dd->gateway->path = g_strdup("/?encoding=json&v=6");
     }
 
     g_match_info_free(match);
